components/camel-kudu/src/main/java/org/apache/camel/component/kudu/KuduEndpoint.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.kudu;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.camel.support.DefaultEndpoint;
+import org.apache.kudu.client.KuduClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Represents a Kudu endpoint. A kudu endpoint allows you to interact with
+ * <a href="https://kudu.apache.org/">Apache Kudu</a>, a free and open source
+ * column-oriented data store of the Apache Hadoop ecosystem.
+ */
+@UriEndpoint(firstVersion = "2.25.0",
+    scheme = "kudu",
+    title = "Apache Kudu", syntax = "kudu:host:port/tableName",
+    label = "cloud,database,iot")
+public class KuduEndpoint extends DefaultEndpoint {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KuduEndpoint.class);
+    private KuduClient kuduClient;
+
+    @Metadata(required = true, description = "Host of the server to connect to")
+    private String host;
+
+    @UriPath(description = "Connection string to Kudu")
+    private String uri;
+
+    @Metadata(required = true)
+    private String port;
+
+    @UriParam(description = "Operation to perform")
+    private KuduOperations operation;
+
+    @Metadata(description = "Table to connect to")
+    private String tableName;
+
+    public KuduEndpoint(String uri, KuduComponent component) {
+        super(uri, component);
+        this.setUri(uri);
+        this.setEndpointUri(uri);
+        Pattern p = Pattern.compile("^(\\S+)\\:(\\d+)\\/(\\S+)$");
```

Comment:
I can see you are parsing the URI manually, I think is better to delegate that to Camel in order to do it for you. In `KuduComponent`, under `createEndpoint` method , you will notice `Map<String, Object> parameters`, that already parsed by Camel and when you do `setProperties` in `KuduComponent`, actually Camel will set all the properties for you, plus it will generate the correct docs for you for these properties. However you might need to adjust your endpoint structure e.g: `kudo:operation?port=&host=&tableName=` or whatever struture you like as long you have one UriPath, as well as adding the correct annotations e.g: `UriParam` and `UriPath`, I'd advise you to take a look at [SolrEndpoint](https://github.com/apache/camel/blob/master/components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrEndpoint.java#L43) and [FileWatch](https://github.com/apache/camel/blob/master/components/camel-file-watch/src/main/java/org/apache/camel/component/file/watch/FileWatchEndpoint.java#L41) as they have very similar structure to your component

Reply:
Isn't it more "human understandable" to use the uri part for the host, port and table? I was checking componentes like:
https://camel.apache.org/components/latest/couchdb-component.html#_uri_format
https://camel.apache.org/components/latest/pg-replication-slot-component.html

That are databases that uses the `host:port/schema?option` form. As someone who just landed on camel, to me it makes sense to have a classic url form. But if you tell me the "standard" in camel is on parameters, I change it.

Reply:
Aha I see, then you can have what you think what it makes sense, as long you can delgate the params parsing to Camel and just parse the path type manually like here: https://github.com/apache/camel/blob/master/components/camel-couchdb/src/main/java/org/apache/camel/component/couchdb/CouchDbEndpoint.java#L45

Reply:
@Delawen Sorry I missed that, actually what you have now is perfectly fine from view. Just add `@UriPath` annotations on `host`, `port` and `tableName` so Camel can generate the correct metadata for the docs